src/EditorComponent.jsx

```diff
@@ -53,119 +53,52 @@ const EditorComponent = props => {
+    const cmdSeekToTimestamp = makeCmdSeekToTimestamp(doCommand);
 
-      if (dispatch) {
-        dispatch(state.tr.replaceSelectionWith(text, false));
-      }
+    window.addEventListener('message', extensionResponseHandler, false);
 
+    const cmdDebugPrint = (state, dispatch) => {
       return true;
     };
 
-    // Editor command that simply tells the extension to capture the current frame. Does not apply any
-    // transaction to the state.
-    const tellExtensionToCaptureFrame = (_state, _dispatch) => {
-      const { videoId } = parentApp.currentVideoInfo();
-      if (videoId) {
-        window.frames[0].postMessage({ type: AppConfig.CaptureCurrentFrameMessage }, '*');
-      }
-      return true;
-    };
-
-    // Listener that checks receiving the data url from extension.
-    const gotResponseFromExtension = e => {
-      if (e.data.type !== AppConfig.CaptureCurrentFrameResponse || !editorView.current) {
-        return;
-      }
-
-      const { data } = e;
-      const { dataURL: source, width: origWidth, height: origHeight } = data;
-
-      // Initial width is same as that of original image width
-      const outerWidth = `${origWidth}px`;
-
-      const view = editorView.current;
-      const { state } = view;
-
-      const { $from } = state.selection;
-      const index = $from.index;
-
-      if (!$from.parent.canReplaceWith(index, index, ImageNodeType)) {
-        return;
-      }
-
-      const videoTime = floorOrZero(doCommand('currentTime'));
-
-      const newState = state.apply(
-        state.tr.replaceSelectionWith(
-          ImageNodeType.create({
-            source,
-            outerWidth,
-            videoTime,
-            origWidth,
-            origHeight,
-          })
-        )
-      );
-
-      view.updateState(newState);
-    };
-
-    window.addEventListener('message', gotResponseFromExtension, false);
-
     // Pause input rule. Typing "#." will toggle pause.
     const ToggleVideoPauseInputRule = new InputRule(/#\.$/, (state, match, start, end) => {
       doCommand('togglePause');
       return state.tr.insertText('', start, end);
     });
 
+    const cmdFunctions = {
+      toggle_pause: () => doCommand('togglePause'),
+      mark_selection_as_timestamp: cmdToggleTimestampMark,
+      put_timestamp: cmdPutTimestampText,
+      seek_to_timestamp: cmdSeekToTimestamp,
+      capture_frame: cmdTellExtensionToCaptureFrame,
+      debug_print: cmdDebugPrint,
+      turn_text_to_timestamp: cmdTurnTextToTimestamp,
+    };
+
+    // Map from keycode to command function
+    const keymapObject = Object.fromEntries(
+      Object.entries(commandNameOfKeycode).map(([keycode, cmdName]) => {
+        return [keycode, cmdFunctions[cmdName]];
+      })
+    );
+
+    const autosavePlugin = new Plugin({
+      view: editorView => new AutosavePlugin(editorView, videoId),
+    });
+
+    const content = document.getElementById('__content__');
```

Comment:
Whats this for?

Reply:
Removing. It was there for test content.